vortex-expr/src/select.rs

```diff
@@ -36,50 +29,84 @@ impl VortexExpr for Select {
     }
 
     fn evaluate(&self, batch: &Array) -> VortexResult<Array> {
-        let Ok(st) = StructArray::try_from(batch) else {
-            vortex_bail!("Argument must be a struct")
-        };
-        let fields = self
-            .columns
-            .iter()
-            .map(|c| c.evaluate(batch))
-            .collect::<VortexResult<Vec<_>>>()?;
-        let names = self
-            .columns
-            .iter()
-            .map(|c| match c.field() {
-                Field::Name(n) => n.as_str().into(),
-                Field::Index(i) => st.names()[*i].clone(),
-            })
-            .collect();
+        batch.with_dyn(|s| {
+            let st = s
+                .as_struct_array()
+                .ok_or_else(|| vortex_err!("Not a struct array"))?;
+            match self {
+                Select::Include(f) => st.project(f),
+                Select::Exclude(e) => {
+                    let normalized_exclusion = e
+                        .iter()
+                        .map(|ef| match ef {
+                            Field::Name(n) => Ok(n.as_str()),
+                            Field::Index(i) => st
+                                .names()
+                                .get(*i)
+                                .map(|s| &**s)
+                                .ok_or_else(|| vortex_err!("Column doesn't exist")),
+                        })
+                        .collect::<VortexResult<Vec<_>>>()?;
+                    let included_names = st
+                        .names()
+                        .iter()
+                        .filter(|f| !normalized_exclusion.contains(&&***f))
```

Comment:
I guess we're assuming that normalized_exclusion is very short...? because contains on Vec is obviously O(N)

Reply:
This is being lazy. We can make a hashset here